src/arcade_collection/analysis/calculate_graph_metrics.py

```diff
@@ -0,0 +1,197 @@
+from typing import Any, Optional
+from dataclasses import dataclass
+
+from prefect import task
+import networkx as nx
+import igraph as ig
+import numpy as np
+
+
+@dataclass
+class GraphMetrics:
+    """
+    Dataclass for storing graph metrics
+    """
+
+    nodes: int
+    edges: int
+    radius: float
+    diameter: float
+    avg_eccentricity: float
+    avg_shortest_path: float
+    avg_in_degrees: float
+    avg_out_degrees: float
+    avg_degree: float
+    avg_clustering: float
+    avg_closeness: float
+    avg_betweenness: float
+    components: int
+    name: Optional[str] = None
+
+    def __getitem__(self, item: str) -> Any:
+        return getattr(self, item)
+
+
+@task
+def calculate_graph_metrics(edges: list[list[int]], weights: list[float] = None) -> GraphMetrics:
```

Comment:
we need to confirm that this will take in a `list[list[Any]`, or at the very least `list[list[tuple[float,...] | int]]`. The likely vertices will be coordinates in 3D space, so at the very least we need to confirm that both igraph and networkx will work with that.